python/delta_sharing/tests/test_reader.py

```diff
@@ -183,3 +186,54 @@ def list_files_in_table(
     expected = pd.concat([expected1, expected2])[["a", "b", "c"]].reset_index(drop=True)
 
     pd.testing.assert_frame_equal(pdf, expected)
+
+
+@pytest.mark.skipif(not ENABLE_INTEGRATION, reason=SKIP_MESSAGE)
+def test_to_pandas_empty(rest_client: DataSharingRestClient):
+    class RestClientMock:
+        def list_files_in_table(
+            self,
+            table: Table,
+            *,
+            predicateHints: Optional[Sequence[str]] = None,
+            limitHint: Optional[int] = None,
+        ) -> ListFilesInTableResponse:
+            assert table == Table("table_name", "share_name", "schema_name")
+
+            metadata = Metadata(
+                schema_string=(
+                    '{"fields":['
+                    '{"metadata":{},"name":"a","nullable":true,"type":"boolean"},'
+                    '{"metadata":{},"name":"b","nullable":true,"type":"byte"},'
+                    '{"metadata":{},"name":"c","nullable":true,"type":"short"},'
+                    '{"metadata":{},"name":"d","nullable":true,"type":"integer"},'
+                    '{"metadata":{},"name":"e","nullable":true,"type":"long"},'
+                    '{"metadata":{},"name":"f","nullable":true,"type":"float"},'
+                    '{"metadata":{},"name":"g","nullable":true,"type":"double"},'
+                    '{"metadata":{},"name":"h","nullable":true,"type":"decimal(5,2)"},'
+                    '{"metadata":{},"name":"i","nullable":true,"type":"string"},'
+                    '{"metadata":{},"name":"j","nullable":true,"type":"binary"},'
+                    '{"metadata":{},"name":"k","nullable":true,"type":"timestamp"},'
+                    '{"metadata":{},"name":"l","nullable":true,"type":"date"},'
+                    '{"metadata":{},"name":"m","nullable":true,"type":{"type":"array",'
+                    '"elementType":"string","containsNull":true}},'
+                    '{"metadata":{},"name":"n","nullable":true,"type":{"type":"struct","fields":'
+                    '[{"name":"foo","type":"string","nullable":true,"metadata":{}},'
+                    '{"name":"bar","type":"integer","nullable":true,"metadata":{}}]}},'
+                    '{"metadata":{},"name":"o","nullable":true,"type":{"type":"map",'
+                    '"keyType":"string","valueType":"integer","valueContainsNull":true}}'
+                    '],"type":"struct"}'
+                )
+            )
+            add_files: Sequence[AddFile] = []
+            return ListFilesInTableResponse(protocol=None, metadata=metadata, add_files=add_files)
+
+    reader = DeltaSharingReader(
+        Table("table_name", "share_name", "schema_name"), RestClientMock()  # type: ignore
+    )
+    pdf = reader.to_pandas()
+
+    reader = DeltaSharingReader(Table(name="table7", share="share1", schema="default"), rest_client)
+    expected = reader.to_pandas().iloc[0:0]
```

Comment:
Shouldn't the `expected` be created in local? Otherwise, this is trying to test `DeltaSharingReader` using `DeltaSharingReader`. I think it's weird.

The integration test should be at `test_load` in `test_delta_sharing.py`.

Reply:
The objective was to make sure the empty table schema is consistent with the non-empty ones, so I added a reference table to compare with. If we make our own reference schema, we can't be sure our reference schema matches the real one when a table is actually read, hence we're using `DeltaSharingReader`.

Although we're using the class to test itself, we're using the non-empty table behavior to test the empty table behavior, so it's a different logical path.

Reply:
I can make a local copy of the parquet file for the table so that this test isn't an integration test, but it's difficult to be sure we have the correct schema without actually reading a table.

Reply:
I see. I'm still feeling weird, though.
Anyway the behavior looks fine if the `table7` has a proper schema.
I'd leave it to @zsxwing for this.

Reply:
I'm okay with this in the short term. Eventually we should make the server support local file system and use all of these tables locally.